src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -896,12 +897,27 @@ def compile(
             self.project.compile(model, no_dedent=False)
             return
 
-        # Get the service targeted by the compile
-        service = self.get_service(service_id)
+        # Sort out the type variance of service_id
+        if isinstance(service_id, (str, uuid.UUID)):
+            # strings are also sequences
+            service_ids = str(service_id)
+        elif isinstance(service_id, typing.Sequence):
+            service_ids = " ".join(str(i) for i in service_id)
+        else:
+            raise TypeError(f"Unexpected argument type for service_id, got {service_id} ({type(service_id)})")
+
+        # Make sure all instances exist in the inventory
+        for service_id in service_ids.split(" "):
+            service = self.get_service(service_id)
 
         env: dict[str, str] = {}
-        env[inmanta_lsm.const.ENV_INSTANCE_ID] = str(service_id)
-        env[inmanta_lsm.const.ENV_INSTANCE_VERSION] = str(service.version)
+        env[inmanta_lsm.const.ENV_INSTANCE_ID] = service_ids
+
+        if validation:
+            if len(service_ids.split(" ")) != 1:
+                raise Exception(f"when performing a validation compile, only one service id can be passed, got {service_ids}")
```

Comment:
```suggestion
                raise Exception(f"when performing a validation compile, only one service id can be passed, got {repr(service_ids)}")
```
That will make it a lot easier to read, especially if for some reason there is a whitespace at the start or the end of the string